core/src/main/java/org/jdbi/v3/core/statement/ColonPrefixSqlParser.java

```diff
@@ -39,23 +37,22 @@
  * </p>
  */
 public class ColonPrefixSqlParser implements SqlParser {
-    private final Map<String, ParsedSql> cache = Collections.synchronizedMap(new WeakHashMap<>());
+    private static final String CACHE_PREFIX = "ColonPrefixSqlParser:";
+
+    private static final JdbiCache<String, ParsedSql> PARSED_SQL_CACHE =
+        JdbiCaches.declare(rawSql -> CACHE_PREFIX + rawSql, ColonPrefixSqlParser::internalParse);
```

Comment:
You don't need this prefix. Each `JdbiCache` obtained through `JdbiCaches.declare()` is distinct and independent.

Reply:
You are right, I did not notice that the instance of JdbiCache itself is also used as a key in the global cache. Will fix in a second.